test/integration/dialects/sqlite/dao.test.js

```diff
@@ -61,26 +64,242 @@ if (dialect === 'sqlite') {
           projects: [
             { dateField: new Date(1990, 5, 5) }
           ]
-        }, { include: [this.Project]}).bind(this).then(function () {
+        }, { include: [this.Project]}).then(() => {
           return this.User.findAll({
             include: [this.Project]
           }).get(0);
-        }).then(function (user) {
+        }).then(user => {
           expect(user.projects[0].get('dateField')).to.be.an.instanceof(Date);
           expect(user.projects[0].get('dateField')).to.equalTime(new Date(1990, 5, 5));
         });
       });
     });
 
+    describe('json', function() {
```

Comment:
We should be able to run these tests for any dialect that supports JSON, right ?